ulid/__init__.py

```diff
@@ -22,13 +22,17 @@
 __version__ = version("python-ulid")
 
 
-class validate_type:  # noqa: N801
-    def __init__(self, *types: Any) -> None:
+T = TypeVar("T", bound=type)
+ReturnType = TypeVar("ReturnType")
```

Comment:
nit: In my experience, type variables have a rather concise name so that I think simply `R` would also work here.

Reply:
Sounds good.  Stole the `ReturnType` from some of the ParamSpec examples, but `R` works equally well for me.